games-emulation/es-de/es_de-3.1.0.recipe

```diff
@@ -0,0 +1,94 @@
+SUMMARY="Gaming frontend for your multi-platform collection"
+DESCRIPTION="ES-DE (EmulationStation Desktop Edition) is a frontend for browsing and launching games from your \
+multi-platform collection. It comes preconfigured for use with a large selection \
+of emulators and game engines."
+HOMEPAGE="https://es-de.org"
+COPYRIGHT="2024 Northwestern Software AB"
+LICENSE="MIT"
+REVISION="1"
+srcGitRev="842c9966eb73efb3436a4df2dfdd66063ce7361c"
+SOURCE_URI="https://gitlab.com/es-de/emulationstation-de/-/archive/$srcGitRev/emulationstation-de-$srcGitRev.tar.gz"
+CHECKSUM_SHA256="52c820beddba7e08014f589120c087d6b84b79a88b6213bf13a4e025fd728285"
+SOURCE_FILENAME="emulationstation-de-$portVersion-$srcGitRev.tar.gz"
+SOURCE_DIR="emulationstation-de-$srcGitRev"
+ADDITIONAL_FILES="es-de.rdef.in"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+        es_de = $portVersion
+        app:es_de = $portVersion
+        "
+
+REQUIRES="
+        haiku$secondaryArchSuffix
+        lib:libGL$secondaryArchSuffix
+        lib:libavcodec$secondaryArchSuffix
+        lib:libavformat$secondaryArchSuffix
+        lib:libavutil$secondaryArchSuffix
+        lib:libcurl$secondaryArchSuffix
+        lib:libfreeimage$secondaryArchSuffix
+        lib:libfreetype$secondaryArchSuffix
+        lib:libharfbuzz$secondaryArchSuffix
+        lib:libgit2$secondaryArchSuffix
+        lib:libicuuc$secondaryArchSuffix
+        lib:libintl$secondaryArchSuffix
+        lib:libpoppler$secondaryArchSuffix
+        lib:libpugixml$secondaryArchSuffix
+        lib:libSDL2_2.0$secondaryArchSuffix
+        "
+
+BUILD_REQUIRES="
+        haiku${secondaryArchSuffix}_devel
+        devel:libGL$secondaryArchSuffix
+        devel:libavcodec$secondaryArchSuffix
+        devel:libavformat$secondaryArchSuffix
+        devel:libavutil$secondaryArchSuffix
+        devel:libcurl$secondaryArchSuffix
+        devel:libfreeimage$secondaryArchSuffix
+        devel:libfreetype$secondaryArchSuffix
+        devel:libharfbuzz$secondaryArchSuffix
+        devel:libgit2$secondaryArchSuffix
+        devel:libicuuc$secondaryArchSuffix
+        devel:libintl$secondaryArchSuffix
+        devel:libpoppler$secondaryArchSuffix
+        devel:libpugixml$secondaryArchSuffix
+        devel:libSDL2_2.0$secondaryArchSuffix
+        "
+
+BUILD_PREREQUIRES="
+        cmd:cmake
+        cmd:g++$secondaryArchSuffix
+        cmd:ld$secondaryArchSuffix
+        cmd:make
+        cmd:msgfmt$secondaryArchSuffix
+        cmd:pkg_config$secondaryArchSuffix
+        "
+
+BUILD()
+{
+        cmake $cmakeDirArgs -DCMAKE_BUILD_TYPE=Release .
```

Comment:
Maybe replace `$cmakeDirArgs` with just `-DCMAKE_INSTALL_PREFIX=$prefix`. The other variables aren't used and cause a warning from cmake.

Reply:
It doesn't build on Haiku Nightly I think, last time I checked there were multiple issues there due to broken dependencies and whatnot. It only builds on R1/beta5. R1/beta4 also didn't work in the past as there were issues with several of the dependencies.

There are also several other issues on Haiku Nightly that makes ES-DE unusable, such as gitlib2 crashing the application and FFmpeg not working correctly. It's really only feasible to use ES-DE on R1/beta5, but I'm not sure if this can be restricted somehow in the recipe so people don't attempt to install and use it on the Nightly releases?

I've anyway removed $cmakeDirArgs and replaced it with -DCMAKE_INSTALL_PREFIX=$prefix as per your recommendation.

Reply:
I actually tried to build on Haiku Nightly with the latest updates applied and it did work so I wonder what is different in your build environment? And libgit2 also doesn't crash the application anymore and FFmpeg also seems to run correctly, so something was obviously fixed in the last two weeks or so which is very good! These are anyway the commands I run to build the package:

```
cd ~
git clone https://github.com/leonstyhre/haikuports.git --depth=50
pkgman install haikuporter
cp /boot/system/settings/haikuports.conf  ~/config/settings/
haikuporter --clean es_de
haikuporter -S --no-source-packages --get-dependencies -j4 es_de
cp ~/haikuports/packages/es_de-3.1.0-1-x86_64.hpkg /boot/system/packages```

Reply:
Ah, I think I used the wrong version of libgit2. It defaulted to version 0.25.